src/hooks/useNotificationTypes.tsx

```diff
@@ -1026,6 +1028,157 @@ export const notificationTypes: NotificationTypeConfig[] = [
       }, [customNotifications, trigger]);
     },
   },
+  {
+    type: NotificationType.DepositAddressEvents,
+    useTrigger: ({ trigger }) => {
+      const stringGetter = useStringGetter();
+      const { dydxAddress } = useAccounts();
+      const { usdcAmount } = useAppSelector(BonsaiCore.account.balances.data);
+
+      const { data: depositStatus } = useDepositStatus();
+
+      // refs to track previous values across renders
+      const prevDepositIdsRef = useRef<Set<string>>(new Set());
+      const prevWalletBalanceRef = useRef<string | undefined>(undefined);
+      const lastNotificationTimeRef = useRef<number>(0);
+
+      const NOTIFICATION_DEBOUNCE = 5 * timeUnits.second; // 5 seconds
+
+      // Stage 1: Watch for deposit from backend
+      useEffect(() => {
+        if (!dydxAddress || !depositStatus?.deposits.results) return;
+
+        const incomingDeposits = depositStatus.deposits.results;
+        const now = Date.now();
+
+        const recentDeposits = incomingDeposits.filter((deposit) => {
+          // Handle both timestamp formats
+          const depositTimestamp =
+            typeof deposit.created_at === 'number'
+              ? deposit.created_at
+              : new Date(deposit.created_at).getTime();
+
+          const depositAge = now - depositTimestamp;
+
+          return depositAge >= 0 && depositAge <= 2 * timeUnits.minute;
+        });
+
+        if (recentDeposits.length === 0) return;
+
+        const prevDepositIds = prevDepositIdsRef.current;
+
+        const newDeposits = recentDeposits.filter((deposit) => !prevDepositIds.has(deposit.id));
+
+        if (newDeposits.length === 0) return;
+
+        newDeposits.forEach((deposit) => {
+          prevDepositIds.add(deposit.id);
+        });
+
+        if (now - lastNotificationTimeRef.current > NOTIFICATION_DEBOUNCE) {
+          trigger({
+            id: `deposit-noble-${Date.now()}`,
```

Comment:
since skip go bypasses `noble` i don't think the id should include it. I think we'd also want the `detected` and `confirmed` id to be the same and possibly use the `deposit.id` instead of `Date.now()`. When you use `Date.now()`, every deposit notification you make will be unique and theres no chance in overwriting the original notif.